src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -109,4 +110,71 @@ def get_date(file):
         except Exception as e:
             print(f"An error occurred: {e}")
 
-        outputs.set_value("corpus_table", output_table)
\ No newline at end of file
+        outputs.set_value("corpus_table", output_table)
+
+class CorpusDistTime(KiaraModule):
+    """
+    This module aggregates a table by day, month or year from a corpus table that contains a date column. It returns the distribution over time, which can be used for display purposes, such as visualization.
+
+    Dependencies:
+    - polars: https://www.pola.rs/
+    - pyarrow: https://arrow.apache.org/docs/python/
+    - duckdb: https://duckdb.org/
+    
+    """
+
+    _module_type_name = "topic_modelling.time_dist"
+
+    def create_inputs_schema(self):
+        
+        return {
+             "periodicity": {
+                "type": "string",
+                "doc": "The desired data periodicity to aggregate the data. Values can be either 'day','month' or 'year'."
+            },
+            "date_col_name": {
+                "type": "string",
+                "doc": "The column that contains the date, a list of compliant formats can be consulted here: https://docs.rs/chrono/latest/chrono/format/strftime/index.html."
+            },
+            "title_col_name": {
```

Comment:
As far as module design goes, this is a very specific requirement. It would require that the data users want to use this with has something like that column, and I don't think we can assume that? What should a user do if their data does not have that? Not use the module? Somehow augment their source table? How? Is it possible to make this more generic, so it will work with any type of column the aggregate can be done against?

Reply:
This module currently aggregates by date to address the needs/requirements expressed for the workflow. If other types of distribution are performed, I would say that it could be treated as a feature request via an issue so that the module is improved/made more generic over time based on the requests. It may very well be the case soon here, as it might be insightful as well (but I need to check with @lorellav first) to have the word count/characters count distribution as well. This is not on the priority list for a first approach though. Would that be ok to start on the current requirements and then re-work the modules as feature requests come in?

Reply:
Right, so this is specific to this notebook? And the module would never appear in any value lineage? Because in that case there's a question whether it needs to be a module at all, or could just be some utility method or put into the notebook directly?

Anyway, I guess that is just one of the cases that falls in my overall concern about how to design a module ecosystem and module interfaces, so I guess we can just leave it as is.

Reply:
This is not necessarily specific to this notebook, as this was identified as something needed to the workflow, and the use case of having a date present within the file name (extracted at the previous step and added as a column) is not marginal (just one example: data obtained via Library of Congress).
The module would appear in value lineage, as this operation does process the initial data. In my opinion, it should be a module, but please feel free to raise the concern at next dev meeting as there may be other opinions on the matter. I will resolve this part of the conversation here for now, but happy to modify, if there are more info about what should or should not be a module.